src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/RandomStressTest/RandomizerPool.cs

```diff
@@ -293,6 +292,14 @@ public void Dispose()
                 }
             }
 
+            /// <summary>
+            /// Disposes the scope. 
+            /// </summary>
+            public void Dispose()
+            {
+                CheckCurrentRandomizerState();
```

Comment:
Why do we need delegated call?

Reply:
Also, why are we throwing an exception at all from Dispose()? Dispose() generally should never throw...

Reply:
Yeah although this is a test - not written well it seems.

Reply:
so, here what I did since we are testing, I have replaced the exception with `Trace.Assert` to check for the required the condition. It was really time consuming following the logic and changing the pattern.